lavalink/client.py

```diff
@@ -203,11 +229,18 @@ def add_node(self, host: str, port: int, password: str, region: str,
         :param event:
             The event to dispatch to the hooks.
         """
-        for hook in self._event_hooks:
-            try:
+        unknown_events = self._event_hooks.get(None) or []
+        registered_events = self._event_hooks.get(event) or []
+
+        event_hooks = unknown_events + registered_events
+
+        try:
+            for hook in event_hooks:
                 if inspect.iscoroutinefunction(hook):
                     await hook(event)
```

Comment:
`asyncio.ensure_future()`

Reply:
I need clearing up on this, on why you might want that.

Reply:
Adding it to queue instead of waiting for it to finish. If someones hook takes longer it could stall the ws loop since this func is awaited (if I'm not mistaken)

Reply:
Example use of async for would be when you're pulling pages of some list from some api.